### PR TITLE
Building challanges

### PR DESCRIPTION
**Pull request by linhmillion**
I'm done with your a part of index challenge

### CODE REVIEW

File: app/controllers/challenges_controller.rb [hunk 1]
```diff
@@ -1,2 +1,49 @@
...
+      @challenges = Challenge.search_by_score_category_name(params[:query])
+    end
+    respond_to do |format|
+      format.html # Follow regular flow of Rails
+      format.text { render partial: 'challenges/list', locals: { challenges: @challenges }, formats: [:html] }
+    end
+  end
+
+  def show
+    @challenge = Challenge.find(params[:id])
```
Comment (daanbm) [inline app/controllers/challenges_controller.rb:15]: can we add this to before_action set_challenge?

File: app/controllers/challenges_controller.rb [hunk 2]
```diff
@@ -1,2 +1,49 @@
...
+  # def destroy
+  # end
+
+  private
+
+  def group_params
+    params.require(:challenge).permit(:name, :description, :category)
+  end
+
+  def set_group
```
[the remark this answers is not in the document]
Comment (daanbm) [reply app/controllers/challenges_controller.rb:46]: do we actually use it? If not, please delete

File: app/controllers/challenges_controller.rb [hunk 3]
```diff
@@ -1,2 +1,49 @@
...
+  def new
+    @challenge = Challenge.new
+  end
+
+  def create
+    @challenge = Challenge.new(challenge_params)
+    if @challenge.save
+      redirect_to challenges_path
+    else
+      render :new
```
Comment (daanbm) [inline app/controllers/challenges_controller.rb:27]: does this path exist?

File: app/views/challenges/show.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<h1>Here is all he changllenges stands</h1>
```
Comment (daanbm) [inline app/views/challenges/show.html.erb:1]: Challenges*

File: app/controllers/challenges_controller.rb (hunk 3 above)
Comment (linhmillion) [reply app/controllers/challenges_controller.rb:27]: yes, but i haven't done page create challenge yet

File: app/views/challenges/show.html.erb (hunk 4 above)
Comment (linhmillion) [reply app/views/challenges/show.html.erb:1]: i need an other content for this though. Any idea?

File: app/controllers/challenges_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (linhmillion) [reply app/controllers/challenges_controller.rb:46]: ok. i will delete this

File: app/controllers/challenges_controller.rb [hunk 5]
```diff
@@ -1,2 +1,49 @@
...
+
+  #   redirect_to challenges_path
+  # end
+
+  # def destroy
+  # end
+
+  private
+
+  def group_params
```
Comment (VeganBt) [inline app/controllers/challenges_controller.rb:42]: should be `challenge_params`

File: app/views/challenges/_list.html.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<div data-search-challenges-target="list">
+
+  <% challenges.all.each do |challenge| %>
+    <div class="card-challenges">
+      <div class="info-challenge">
+        <h2><%=challenge.name %></h2>
+        <p>Discription: <%= challenge.description %></p>
```
Comment (VeganBt) [inline app/views/challenges/_list.html.erb:7]: Description 😀

**Review by daanbm — APPROVED** [review]

**Review by VeganBt — APPROVED** [review]

**Pull request merged**